src/compiler/crystal/compiler.cr

```diff
@@ -240,7 +240,7 @@ module Crystal
 
     private def codegen_single_unit(unit, target_triple, multithreaded)
       unit.llvm_mod.target = target_triple
-      ifdef x86_64
+      if target_triple =~ /^x86_64/
```

Comment:
@ssvb @ysbaddaden I understand the issue now. You are right, `ifdef` shouldn't be used inside the compiler. I just [pushed this commit](https://github.com/crystal-lang/crystal/commit/972c22beac130cfbd76ee251248555c0ee02deb9).

At first I thought: if you aren't cross-compiling and you are on a 64-bits machine then you want 64-bits executables. But it never occurred to me that you could have a 64-bits compiler inside a 32-bits machine.

Do you think the above commit solves the problem?